### PR TITLE
[DF-379] Fix validation of the job publication page

### PR DESCRIPTION
**Pull request by wwebec**

### CODE REVIEW

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 1]
```diff
@@ -63,7 +63,27 @@ import { CurrentSelectedCompanyService } from '../../../shared/company/current-s
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class JobPublicationToolComponent implements OnInit, OnDestroy {
-    readonly APPLICATION_ADDITIONAL_INFO_MAX_LENGTH = 240;
+    readonly APPLICATION_PHONE_NUMBER_MAX_LENGTH = 20;
+    readonly APPLICATION_PAPER_APPLICATION_ADDRESS_MAX_LENGTH = 255;
```
Comment (cjpsycos) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: Why you didn't do it on the avam release branch? now you have to do it twice

applyChannel.rawPostAddress = 255
applyChannel.postAddress.city = 100
applyChannel.postAddress.countryIsoCode = 2
applyChannel.postAddress.houseNumber = 10
applyChannel.postAddress.name = 255
applyChannel.postAddress.postalCode = 10
applyChannel.postAddress.postOfficeBoxCity = 100
applyChannel.postAddress.postOfficeBoxNumber = 20
applyChannel.postAddress.postOfficeBoxPostalCode = 10
applyChannel.postAddress.street = 60

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 2]
```diff
@@ -63,7 +63,27 @@ import { CurrentSelectedCompanyService } from '../../../shared/company/current-s
...
+    readonly PUBLIC_CONTACT_PHONE_NUMBER_MAX_LENGTH = 255;
```
Comment (cjpsycos) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: mail missing with 50
phone is 20

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 3]
```diff
@@ -63,7 +63,27 @@ import { CurrentSelectedCompanyService } from '../../../shared/company/current-s
...
+    readonly PUBLIC_CONTACT_LAST_NAME_MAX_LENGTH = 50;
+    readonly PUBLIC_CONTACT_FIRST_NAME_MAX_LENGTH = 50;
+    readonly PUBLIC_CONTACT_SALUTATION_MAX_LENGTH = 3;
+
+    readonly CONTACT_PHONE_NUMBER_MAX_LENGTH = 20;
```
Comment (cjpsycos) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: mail is missing with 50

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 4]
```diff
@@ -63,7 +63,27 @@ import { CurrentSelectedCompanyService } from '../../../shared/company/current-s
...
+    readonly CONTACT_LAST_NAME_MAX_LENGTH = 50;
+    readonly CONTACT_FIRST_NAME_MAX_LENGTH = 50;
+    readonly CONTACT_SALUTATION_MAX_LENGTH = 3;
+
+    readonly COMPANY_STREET_MAX_LENGTH = 50;
+    readonly COMPANY_COUNTRY_CODE_MAX_LENGTH = 2;
+    readonly COMPANY_POSTBOX_ZIP_CODE_MAX_LENGTH = 255;
```
Comment (cjpsycos) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: postOfficeBoxPostalCode is 10

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 5]
```diff
@@ -63,7 +63,27 @@ import { CurrentSelectedCompanyService } from '../../../shared/company/current-s
...
+    readonly COMPANY_ZIP_CODE_MAX_LENGTH = 10;
+    readonly COMPANY_HOUSE_NO_MAX_LENGTH = 10;
+    readonly COMPANY_NAME_MAX_LENGTH = 50;
```
Comment (cjpsycos) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: company name is 255

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts [hunk 6]
```diff
@@ -63,7 +63,27 @@ import { CurrentSelectedCompanyService } from '../../../shared/company/current-s
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class JobPublicationToolComponent implements OnInit, OnDestroy {
-    readonly APPLICATION_ADDITIONAL_INFO_MAX_LENGTH = 240;
+    readonly APPLICATION_PHONE_NUMBER_MAX_LENGTH = 20;
```
Comment (cjpsycos) [inline src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: mail is missing with 50

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 1 above)
Comment (wwebec) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: base changed to release/AVAM-Changes-RFC556-RFC563

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 2 above)
Comment (wwebec) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: email has already [regex validator](https://github.com/alv-ch/jobroom2/blob/b7399b4ce3937e36cdb7501e0cbd4e410815ca44/src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts#L391)

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 4 above)
Comment (wwebec) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: there is 255 in [COMPANY_POSTBOX_ZIP_CODE](https://github.com/alv-ch/job-ad-service/blob/9621cece7047c4ebe38c688ea09d2059387f8e80/web/src/main/resources/liquibase/changelog/20180211080000_added_entity_JobAdvertisement.xml#L91)

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 5 above)
Comment (wwebec) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: right, changed

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 3 above)
Comment (wwebec) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: email has already [regex validator](https://github.com/alv-ch/jobroom2/blob/b7399b4ce3937e36cdb7501e0cbd4e410815ca44/src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts#L391)

File: src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts (hunk 6 above)
Comment (wwebec) [reply src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts]: email has already [regex validator](https://github.com/alv-ch/jobroom2/blob/b7399b4ce3937e36cdb7501e0cbd4e410815ca44/src/main/webapp/app/home/tools/job-publication-tool/job-publication-tool.component.ts#L402)

**Pull request merged**